wgsl/index.bs

```diff
@@ -5126,30 +5126,106 @@ If the `sampler_comparison` uses bilinear filtering then the returned value is
 the filtered average of these values, otherwise the comparision result of a
 single texel is returned.
 
-**TODO:**
 
-<pre class='def'>
-`vec4<type> textureLoad(texture_storage_ro_1d      , i32       coords)`
-`vec4<type> textureLoad(texture_storage_ro_2d      , vec2<i32> coords)`
-`vec4<type> textureLoad(texture_storage_ro_1d_array, vec2<i32> coords)`
-`vec4<type> textureLoad(texture_storage_ro_3d      , vec3<i32> coords)`
-`vec4<type> textureLoad(texture_storage_ro_2d_array, vec3<i32> coords)`
-  %32 = OpImageRead %v4float %texture_storage_ro %coords
+### `textureRead` ### {#textureread}
+
+Reads a single texel from a texture without sampling or filtering.
+
+<!-- TODO(ben-clayton): Remove parameter alignment padding -->
+
+```rust
+textureRead(t : texture_1d<T>,                    coords : u32)                                                           -> vec4<T>
+textureRead(t : texture_1d_array<T>,              coords : u32,       array_index : u32)                                  -> vec4<T>
+textureRead(t : texture_2d<T>,                    coords : vec2<u32>)                                                     -> vec4<T>
+textureRead(t : texture_2d<T>,                    coords : vec2<u32>,                    level : u32)                     -> vec4<T>
```

Comment:
so interestingly, re-reading https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-to-load appears to show that `Load` doesn't have a way to select a mipmap level at all (in HLSL). So we should not have that either.

Reply:
From the page you linked:

> Location
> [in] The texture coordinates; the last component specifies the mipmap level

Is there something I'm missing?

Reply:
good note! Sorry, I missed it.
So we need to at least remove the mipmap-level variants from MSAA loads. I.e. the table on that page says `int2` is the coordinate type for 2DMS, so there is no mipmap level

Reply:
> So we need to at least remove the mipmap-level variants from MSAA loads.

Right you are. I did think mipmaps on a multisampled texture was a bit odd. Will remove.

Reply:
Done